src/rust/lib/graph-editor/src/component/visualization/container.rs

```diff
@@ -0,0 +1,174 @@
+///
+/// The API to interact with the visualisation is exposed through the `ContainerFrp`.
+#[derive(Clone,CloneRef,Debug,Shrinkwrap)]
+#[allow(missing_docs)]
+pub struct Container {
+    // The internals are split into two structs: `ContainerData` and `ContainerFrp`. The
+    // `ContainerData` contains the actual data and logic for the `Container`. The `ContainerFrp`
+    // contains the FRP api and network. This split is required to avoid creating cycles in the FRP
+    // network: the FRP network holds `Rc`s to the `ContainerData` and thus must not live in the
+    // same struct.
+
+    #[shrinkwrap(main_field)]
+        data : Rc<ContainerData>,
+    pub frp  : ContainerFrp,
+}
+
+/// Internal data of a `Container`.
+#[derive(Debug,Clone)]
+#[allow(missing_docs)]
+pub struct ContainerData {
+    logger         : Logger,
+    display_object : display::object::Instance,
+    size           : Cell<Vector2<f32>>,
+    visualization  : RefCell<Option<Visualization>>,
+}
+
+impl ContainerData {
+    /// Set whether the visualisation should be visible or not.
+    pub fn set_visibility(&self, is_visible:bool) {
+        if let Some(vis) = self.visualization.borrow().as_ref() {
+            if is_visible {
+                vis.display_object().set_parent(&self.display_object);
+            } else {
+                vis.display_object().unset_parent();
+            }
+        }
+    }
+
+    /// Indicates whether the visualisation is visible.
+    fn is_visible(&self) -> bool {
+        if let Some(vis) = self.visualization.borrow().as_ref() {
+            vis.display_object().has_parent()
+        } else {
+            false
+        }
+    }
+
+    /// Toggle visibility.
+    fn toggle_visibility(&self) {
+        self.set_visibility(!self.is_visible())
+    }
+
+    /// Update the content properties with the values from the `ContainerData`.
+    ///
+    /// Needs to called when a visualisation has been set.
+    fn init_visualisation_properties(&self) {
+        let size         = self.size.get();
+        if let Some(vis) = self.visualization.borrow().as_ref() {
+            vis.set_size(size);
+        };
+        self.set_visibility(true);
+    }
+
+    /// Set the visualization shown in this container..
+    fn set_visualisation(&self, visualization:Visualization) {
+        visualization.display_object().set_parent(&self.display_object);
+        self.visualization.replace(Some(visualization));
+        self.init_visualisation_properties();
+    }
+}
+
+impl Container {
+    /// Constructor.
+    pub fn new() -> Self {
+        let logger         = Logger::new("visualization");
+        let visualization  = default();
+        let size           = Cell::new(Vector2::new(100.0, 100.0));
+        let display_object = display::object::Instance::new(&logger);
+        let data           = ContainerData {logger,visualization,size,display_object};
+        let data           = Rc::new(data);
+        let frp            = default();
+
```

Comment:
Lets not use nlns in such places, ok? I know this is very subjective, but we agreed some time ago to make the code minimal vertically. Sometimes it makes sense, of course - when you want to divide it into some related blocks, but here you are just calling init.